python/lsst/afw/cameraGeom/testUtils.py

```diff
@@ -419,20 +419,29 @@ def makeTestRepositoryItems(self, isLsstLike=False):
 
 @inTestCase
 def compare2DFunctions(self, func1, func2, minVal=-10, maxVal=None, nVal=5):
-    """Compare two Point2D(Point2D) functions by evaluating them over a
+    """Compare two Point2D(list(Point2D)) functions by evaluating them over a
     range of values.
+
+    Notes
+    -----
+    Assumes the functions can be called with ``list[Point2D]`` and return
+    ``list[Point2D]``.
     """
     if maxVal is None:
         maxVal = -minVal
     dVal = (maxVal - minVal) / (nVal - 1)
+    points = []
     for xInd in range(nVal):
         x = minVal + (xInd * dVal)
         for yInd in range(nVal):
             y = minVal + (yInd * dVal)
             fromPoint = lsst.geom.Point2D(x, y)
-            res1 = func1(fromPoint)
-            res2 = func2(fromPoint)
-            self.assertPairsAlmostEqual(res1, res2)
+            points.append(fromPoint)
+
+    vres1 = func1(points)
+    vres2 = func2(points)
+    for res1, res2 in zip(vres1, vres2):
+        self.assertPairsAlmostEqual(res1, res2)
```

Comment:
@erykoff I tried using `self.assertPairListsAlmostEqual(vres1, vres2)` but it ends up slowing things down by half a second for each test. It seems odd that `assertPairsAlmostEqual` in a loop is faster than `assertPairListsAlmostEqual`.

Reply:
That is strange that it's slower, but looking at the code for `assertPairListsAlmostEqual` https://github.com/lsst/geom/blob/f0d0fc72e121e9654f877923b359f832be524fe5/python/lsst/geom/testUtils.py#L141 I can see why it isn't faster since it just loops over all the pairs anyway.

Reply:
Maybe I should use the vectored one and hope that someday the vectored one becomes faster?

Reply:
It's not vectored ... and I wouldn't hold my breath.

Reply:
(I meant vectored in the sense that it takes everything in one call and so could be vectorized internally in the future whereas me looping over each point is guaranteed never to become vectorized)